scm/pipeline.go

```diff
@@ -0,0 +1,106 @@
+package scm
+
+type (
+	ObjectAttributes struct {
+		ID         int
+		IID        int
+		Name       string
+		Ref        string
+		Tag        bool
+		SHA        string
+		BeforeSHA  string
+		Source     string
+		Status     string
+		Stages     []string
+		CreatedAt  string
+		FinishedAt string
+		Duration   int
+		Variables  []Variable
+		URL        string
+	}
+
+	Variable struct {
+		Key   string
+		Value string
+	}
+
+	MergeRequest struct {
```

Comment:
we already have a pull request structure.  we should not need to introduce a new merge request structure